stan/math/prim/fun/as_value_array_or_scalar.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef STAN_MATH_PRIM_FUN_AS_VALUE_ARRAY_OR_SCALAR_HPP
+#define STAN_MATH_PRIM_FUN_AS_VALUE_ARRAY_OR_SCALAR_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/as_array_or_scalar.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Extract the value from an object. For eigen types and `std::vectors`
+ * convert to an eigen array and for scalars return a scalar.
+ * @tparam T A stan scalar, eigen vector, or `std::vector`.
+ * @param v Specified value.
+ * @return Same value.
+ */
+template <typename T>
+inline auto as_value_array_or_scalar(T&& v) {
+  return value_of(as_array_or_scalar(std::forward<T>(v)));
```

Comment:
@t4c1 so for this where I'm just forwarding the argument along, does this need a holder?

Reply:
Also other random Q, should we be strong inlining these kinds of functions like Eigen does?

Reply:
> @t4c1 so for this where I'm just forwarding the argument along, does this need a holder?

Both `value_of` and `as_array_or_scalar` handle rvalues (I think only functions defined by Eigen do not), so no.

> Also other random Q, should we be strong inlining these kinds of functions like Eigen does?

I don't think so. Only if you can demonstrate a non-negligible speedup if strong inlining them.